src/itinerary_builder.js

```diff
@@ -21,8 +21,12 @@
       var classes = [ 'leaflet-routing-icon', 'lanes'];
       if (!l.valid) classes.push(['invalid']);
 
+      var indication = l.indications.find(function(e) {
+        return e === step.maneuver.modifier;
+      }) || l.indications[0];
+
       var icon;
-      switch (l.indications[0]) {
+      switch (indication) {
```

Comment:
I think I understand that disconnect.

The only clean solution to me seems to be https://github.com/Project-OSRM/osrm-backend/issues/2935.

There seems to be an edge case where "correct_indication !== modifier but incorrect_indication === modifier". I assume that this edge case is rare and an okay tradeoff to accept within this PR.